kivy/core/text/_text_pango.pyx

```diff
@@ -0,0 +1,372 @@
+    cdef FcConfig *fc_config
+
+    # Called explicitly on malloc fail to release as fast as possible
+    # Note: calling a method from __dealloc__ can lead to revived object
+    def __dealloc__(self):
+        if self.metrics:
+            pango_font_metrics_unref(self.metrics)
+        if self.fontdesc:
+            pango_font_description_free(self.fontdesc)
+        if self.layout:
+            g_object_unref(self.layout)
+        if self.context:
+            g_object_unref(self.context)
+        if self.fontmap:
+            g_object_unref(self.fontmap)
+        if self.fc_config:
+            FcConfigDestroy(self.fc_config)
+
+
+# Add a contextcontainer to cache + keep max 64 open fonts
+cdef inline _add_pango_cache(unicode fontid, ContextContainer cc):
+    global kivy_pango_cache, kivy_pango_cache_order
+
+    cdef unicode popid
+    while len(kivy_pango_cache_order) >= 64:
+        popid = kivy_pango_cache_order.pop(0)
+        # FIXME: I think this is safe? but leaving out for now
+        #kivy_pango_cache[popid].__dealloc__()
+        del kivy_pango_cache[popid]
+
+    kivy_pango_cache[fontid] = cc
+    kivy_pango_cache_order.append(fontid)
+
+
+# NOTE: for future, this applies to font selection, irrelevant with one font
+# Or maybe it is relevant, it's not clear. The callback didn't execute with
+# null for data/destroy callback. I was planning on testing with flags
+# packed in a gpointer, but this is still on todo. If anyone knows more
+# about fontconfig and can clarify how this should be done, that'd be great
+
+#cdef _configure_pattern_destroy_data(gpointer data):
+#    print("_configure_pattern_destroy_data()!")
+
+#cdef _configure_pattern_callback(FcPattern *pattern, gpointer data):
+#    cdef unsigned int flags = GPOINTER_TO_UINT(data)
+#    print("_configure_pattern_callback()!")
+#    FcPatternDel(pattern, FC_HINTING)
+#    FcPatternAddBool(pattern, FC_HINTING, ...)
+#    FcPatternDel(pattern, FC_AUTOHINT)
+#    FcPatternAddBool(pattern, FC_AUTOHINT, ...)
+#    FcPatternDel(pattern, FC_HINT_STYLE)
+#    FcPatternAddInteger(pattern, FC_HINT_STYLE, ...)
+#    FcPatternDel(pattern, FC_ANTIALIAS)
+#    FcPatternAddBool(pattern, FC_ANTIALIAS, ...)
+
+
+# Creates a ContextContainer for the font_name_r of the label
+cdef _get_context_container(kivylabel):
+    global kivy_pango_text_direction
+    cdef dict options = kivylabel.options
+    cdef unicode fontid = <unicode>options['font_name_r']
+    if fontid in kivy_pango_cache:
+        return kivy_pango_cache.get(fontid)
+
+    # Creat a new context
+    cdef ContextContainer cc = ContextContainer()
+
+    # Create blank FcConfig (fontconfig), and load the TTF file
+    cc.fc_config = FcInitLoadConfig()
+    cdef bytes filename = options['font_name_r'].encode('UTF-8')
+    if FcConfigAppFontAddFile(cc.fc_config, <FcChar8 *>filename) == FcFalse:
+        Logger.warn("_text_pango: Error loadinging font '{}'".format(filename))
+        cc.__dealloc__()
```

Comment:
Nop, don't call it yourself. Just don't use it anymore, and leave it to GC later (cython will call the __dealloc__ itself). CF http://cython.readthedocs.io/en/latest/src/userguide/special_methods.html#finalization-method-dealloc

Reply:
The fix is just "return" (None implicit, so cc will not be used anymore therefore goes into garbage collection)